support-files/mariadb-systemd-start

```diff
@@ -0,0 +1,44 @@
+#!/bin/bash
+#
+# Scripts to run by MySQL systemd service
+#
+#
+# try to run mysql_install_db and fix perms and SELinux contexts
+
+install_db () {
+    # Note: something different than datadir=/var/lib/mysql requires SELinux policy changes (in enforcing mode)
+    datadir=/var/lib/mysql
+    log=/var/log/mysqld.log
+    coproc mysqldefaults { /usr/bin/my_print_defaults --mysqld ; }
+    while read var; do
+      key=${var%%=*}
+      if [ "${key}" = '--datadir' ]; then
+        datadir=${var#*=}
+      fi
+      if [ "${key}" = '--log-error' ] || [ "${key}" = '--log_error' ]; then
+        log=${var#*=}
+      fi
+    done < /dev/fd/${mysqldefaults[0]}
+
+    # Restore log, dir, perms and SELinux contexts
+    [ -d "$datadir" ] || install -d -m 0755 -omysql -gmysql "$datadir" || exit 1
+    [ -e $log ] || touch $log
+    chmod 0640 $log
+    chown mysql:mysql $log || exit 1
+    if [ -x /usr/sbin/restorecon ]; then
+        /usr/sbin/restorecon "$datadir"
+        /usr/sbin/restorecon $log
+    fi
+
+    # If special mysql dir is in place, skip db install
+    [ -d "$datadir/mysql" ] && exit 0
+
+    # Create initial db
+    /usr/bin/mysql_install_db --rpm --datadir="$datadir" --user=mysql
+    exit 0
```

Comment:
1. ok - can leave to packaging - debian uses postinst for this
2. this is a slight rework of ./rpm-oel/mysql-systemd-start